lib/exfat_dir.c

```diff
@@ -502,28 +502,43 @@ int exfat_lookup_file(struct exfat *exfat, struct exfat_inode *parent,
 	return 0;
 }
 
-static void unix_time_to_exfat_time(time_t unix_time, __u8 *tz, __le16 *date,
-				    __le16 *time, __u8 *time_ms)
+void exfat_calc_dentry_checksum(struct exfat_dentry *dentry,
+				uint16_t *checksum, bool primary)
 {
-	struct tm tm;
-	__u16 t, d;
+	unsigned int i;
+	uint8_t *bytes;
 
-	gmtime_r(&unix_time, &tm);
-	d = ((tm.tm_year - 80) << 9) | ((tm.tm_mon + 1) << 5) | tm.tm_mday;
-	t = (tm.tm_hour << 11) | (tm.tm_min << 5) | (tm.tm_sec >> 1);
+	bytes = (uint8_t *)dentry;
 
-	*tz = 0x80;
-	*date = cpu_to_le16(d);
-	*time = cpu_to_le16(t);
-	if (time_ms)
-		*time_ms = (tm.tm_sec & 1) * 100;
+	*checksum = ((*checksum << 15) | (*checksum >> 1)) + bytes[0];
+	*checksum = ((*checksum << 15) | (*checksum >> 1)) + bytes[1];
+
+	i = primary ? 4 : 2;
+	for (; i < sizeof(*dentry); i++)
+		*checksum = ((*checksum << 15) | (*checksum >> 1)) + bytes[i];
+}
+
+static uint16_t calc_dentry_set_checksum(struct exfat_dentry *dset, int dcount)
```

Comment:
So, Why do you add this functions ? Do you think that it is good to describe in patch description ?